MSAL/src/native_auth/public/MSALNativeAuthUserAccountResult+Internal.swift

```diff
@@ -26,20 +26,45 @@ import Foundation
 
 extension MSALNativeAuthUserAccountResult {
 
-    func getAccessTokenInternal(
-        forceRefresh: Bool,
-        correlationId: UUID?,
-        cacheAccessor: MSALNativeAuthCacheInterface
-    ) async -> MSALNativeAuthCredentialsControlling.RefreshTokenCredentialControllerResponse {
-        let context = MSALNativeAuthRequestContext(correlationId: correlationId)
-        let correlationId = context.correlationId()
-
-        if forceRefresh || self.authTokens.accessToken.isExpired() {
-            let controllerFactory = MSALNativeAuthControllerFactory(config: configuration)
-            let credentialsController = controllerFactory.makeCredentialsController(cacheAccessor: cacheAccessor)
-            return await credentialsController.refreshToken(context: context, authTokens: authTokens, userAccountResult: self)
-        } else {
-            return .init(.success(MSALNativeAuthTokenResult(authTokens: authTokens)), correlationId: correlationId)
+    func getAccessTokenInternal(forceRefresh: Bool = false,
+                                scopes: [String],
+                                correlationId: UUID? = nil,
+                                delegate: CredentialsDelegate) {
+
+        let params = MSALSilentTokenParameters(scopes: scopes, account: account)
+        params.forceRefresh = forceRefresh
+        params.correlationId = correlationId
+
+        guard let config = MSALNativeAuthPublicClientApplication.sharedConfiguration,
+              let challengeTypes = MSALNativeAuthPublicClientApplication.sharedChallengeTypes,
+              let client = try? MSALNativeAuthPublicClientApplication(configuration: config, challengeTypes: challengeTypes)
+        else {
```

Comment:
can you log an error here:
```suggestion
        else {
        MSALLogger.log(
                level: .error,
                context: context,
                format: "Config or challenge types unexpectedly found nil"
            )
```